### ISSUE TITLE
[dev.icinga.com #7971] Ensure that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1241]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/7971

**Created by elippmann on 2014-12-03 11:50:22 +00:00**

Assignee: _(none)_
Status: _Closed_  (closed on _2016-02-19 16:06:31 +00:00_)
Target Version: _Accessibility_
Last Update: _2016-02-19 16:06:31 +00:00 (in Redmine)_

---

http://www.w3.org/TR/WCAG20-TECHS/general.html#G145 wrote:
> The objective of this technique is to make sure that users can read text that is presented over a background. This technique relaxes the 4.5:1 contrast ratio requirement for text that is at least 18 point (if not bold) or at least 14 point (if bold).
> 
> If the background is a solid color (or all black or all white) then the contrast ratio of the larger-scale text can be maintained by making sure that each of the text letters have a 3:1 contrast ratio with the background.
> 
> If the background or the letters vary in relative luminance (or are patterned), then the background around the letters can be chosen or shaded so that the letters maintain a 3:1 contrast ratio with the background behind them even if they do not have that contrast ratio with the entire background.
> 
> The contrast ratio can sometimes be maintained by changing the relative luminance of the letters as the relative luminance of the background changes across the page.
> 
> Another method is to provide a halo around the text that provides the necessary contrast ratio if the background image or color would not normally be sufficiently different in relative luminance.
